### PR TITLE
ADD: Feature/rest api documentation

### LINKED ISSUE #111: Documentation for FileController
Can also add documentation for the `FileController`?

_Originally posted by @Segelzwerg in https://github.com/hhu-propra2/abschlussprojekt-mopse/pull/105#issuecomment-596572817_

### PR DESCRIPTION
**Pull request by Segelzwerg**
closes #111

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -35,3 +35,5 @@ out/
 prod.env
 minio_data/
 !javadoc/
+/build/generated/sources/
+/build/generated/
```
Comment (iTitus) [inline .gitignore]:
```suggestion
build/
!build/generated-snippes/
```
would be more elegant

File: build.gradle [hunk 2]
```diff
@@ -61,6 +62,9 @@ dependencies {
     testImplementation 'org.testcontainers:testcontainers:1.13.0'
     testImplementation 'org.testcontainers:junit-jupiter:1.13.0'
     testImplementation 'com.c4-soft.springaddons:spring-security-test-keycloack-addons:1.2.0'
+    asciidoctor 'org.springframework.restdocs:spring-restdocs-asciidoctor'
+    testImplementation 'org.springframework.restdocs:spring-restdocs-mockmvc'
+
```
Comment (iTitus) [inline build.gradle]:
```suggestion
```

**Comment by Segelzwerg** [comment]
the generated html page wil be added to the github pages

**Comment by Segelzwerg** [comment]
The documentation file `documentation.adoc` is only download. The should be exported as html and presented as a website.

File: build.gradle [hunk 3]
```diff
@@ -93,10 +96,15 @@ tasks.withType(com.github.spotbugs.snom.SpotBugsTask) {
     }
 }
 
+
+import io.franzbecker.gradle.lombok.task.DelombokTask
+import org.asciidoctor.gradle.AsciidoctorTask
```
[the remark this answers is not in the document]
Comment (Segelzwerg) [reply build.gradle]: Oh didn't know, Intellij was suggesting it.

File: build.gradle [hunk 4]
```diff
@@ -106,9 +114,53 @@ task delombok(type: io.franzbecker.gradle.lombok.task.DelombokTask, dependsOn: c
...
+
+asciidoctor {
+    inputs.dir snippetsDir
+    dependsOn test
+}
+
+test {
+    useJUnitPlatform()
+    outputs.dir snippetsDir
+}
```
[the remark this answers is not in the document]
Comment (Segelzwerg) [reply build.gradle]: oh I deleted the above on.

File: README.md [hunk 5]
```diff
@@ -22,6 +22,8 @@ Es ist aufwendig für Lerngruppen oder Kurse Dateien miteinander zu teilen, wenn
 
 ## Dokumentation
 
-Hier kann die Dokumentation gefunden werden: [Dokumentation](/doc/documentation.adoc)
+Hier kann die Dokumentation gefunden werden: [Dokumentation](/doc/html5/documentation.html)
```
Comment (iTitus) [inline README.md]: use the github pages link here

Comment (Segelzwerg) [reply README.md]: that should've been there. I think something didn't got pushed.

File: build.gradle [hunk 6]
```diff
@@ -106,9 +105,53 @@ task delombok(type: io.franzbecker.gradle.lombok.task.DelombokTask, dependsOn: c
     }
 }
 
+ext {
+    snippetsDir = file('build/generated-snippets')
```
Comment (iTitus) [inline build.gradle]:
```suggestion
    snippetsDir = file 'build/generated-snippets'
```
use all the groovy features you can!

**Comment by Segelzwerg** [comment]
> please remove the adocs

will do

File: api/external-api.adoc [hunk 7]
```diff
@@ -3,18 +3,18 @@ Anfrage Gruppenurl::
 
 Anfrage:
 
-include::../build/generated-snippets/index/GroupController/get-group-url/path-parameters.adoc[]
+include::{snippets}/index/GroupController/get-group-url/path-parameters.adoc[]
```
Comment (Segelzwerg) [inline api/external-api.adoc]: The advantage of the previous version is you can run it from command line.

Comment (iTitus) [reply api/external-api.adoc]: docToHtml task didn't really work

**Pull request merged**